src/log.cpp

```diff
@@ -459,13 +459,8 @@ void log_printf(logLevel level, const char* format, ...) {
     log_file.close();
 
     /* Write the log message to the shell */
-    std::cout << msg_string;
+    printf(msg_string.c_str());
   }
-
-
-  /* If the message was a runtime error, exit the program */
-  if (level == ERROR)
-    exit(1);
 }
```

Comment:
I've corrected this issue per a suggestion from the SWIG mailing list. Now when an ``ERROR`` message is printed, it throws a ``logic_error(...)`` exception rather than calling ``exit()``. This allows the IPython Notebook kernel to continue running, but immediately terminates the execution of whatever routine was previously running. Alternatively, it will kill a process running a Python script as is currently done.